lms/djangoapps/instructor/tests/test_certificates.py

```diff
@@ -734,3 +736,159 @@ def test_generate_certificate_exceptions_invalid_user_list_error(self):
+            email='test_student@example.com',
+            first_name='NotEnrolled',
+            last_name='Student'
+        )
+
+        self.not_enrolled_student = UserFactory(
+            username='NotEnrolledStudent',
+            email='nonenrolled@test.com',
+            first_name='NotEnrolled',
+            last_name='Student'
+        )
+        CourseEnrollment.enroll(self.enrolled_user, self.course.id)
+
+        # Global staff can see the certificates section
+        self.client.login(username=self.global_staff.username, password="test")
+
+    def test_create_white_list_exception_record(self):
+        """
+        Happy path test to create a single new white listed record
+        """
+        csv_content = "test_student@example.com,dummy_notes"
+        uploaded_file = SimpleUploadedFile("temp.csv", csv_content)
+        response = self.client.post(self.url, {'students_list': uploaded_file})
+        self.assertEqual(response.status_code, 200)
+        data = json.loads(response.content)
+        self.assertEquals(len(data['errors']), 0)
+        self.assertEquals(len(data['warnings']), 0)
+        self.assertEquals(len(data['success']), 1)
+        self.assertEquals(len(CertificateWhitelist.objects.all()), 1)
+
+    def test_invalid_data_format_in_csv(self):
+        """
+        Try uploading a CSV file with invalid data formats and verify the errors.
+        """
+        csv_content = "test_student@example.com,test,1,USA\n" \
+                      "test_student@example.com,test,1"
+
+        uploaded_file = SimpleUploadedFile("temp.csv", csv_content)
+        response = self.client.post(self.url, {'students_list': uploaded_file})
+        self.assertEqual(response.status_code, 200)
+        data = json.loads(response.content)
+        self.assertEquals(len(data['errors']), 2)
+        self.assertEquals(len(data['warnings']), 0)
+        self.assertEquals(len(data['success']), 0)
+        self.assertEqual(data['errors'][0].get('response'), u'-- Data in row# 1 must have exactly two columns, either '
+                                                            u'an email or a username and notes.')
+
+    def test_file_upload_type_not_csv(self):
+        """
+        Try uploading some non-CSV file and verify that it is rejected
+        """
+        uploaded_file = SimpleUploadedFile("temp.jpg", io.BytesIO(b"some initial binary data: \x00\x01").read())
+        response = self.client.post(self.url, {'students_list': uploaded_file})
+        self.assertEqual(response.status_code, 200)
+        data = json.loads(response.content)
+        self.assertNotEquals(len(data['errors']), 0)
+        self.assertEquals(data['errors'][0]['response'], '-- Make sure that the file you upload is in CSV format with '
+                                                         'no extraneous characters or rows.')
+
+    def test_bad_file_upload_type(self):
+        """
+        Try uploading some non-CSV file and verify that it is rejected
```

Comment:
From docstring on this test it looks like the the one above and this one are doing same thing. Can we change docstring which explains how these tests are different?